api/METRICS.md

```diff
@@ -35,72 +45,260 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
 acc_group_min = accuracy_score_group_min(y_true, y_pred, sensitive_features=sf, **other_kwargs)
 ```
 
-## Proposal
+## Proposal details
 
-*Function signatures*
+The items that are not implemented yet are marked as `[TODO]`.
+
+### Metrics engine
 
 ```python
 group_summary(metric, y_true, y_pred, *, sensitive_features, **other_kwargs)
 # return the group summary for the provided `metric`, where `metric` has the signature
 # metric(y_true, y_pred, **other_kwargs)
 
-make_metric_group_summary(metric)
+make_group_summary(metric) [TODO]
 # return a callable object <metric>_group_summary:
-# <metric>_group_summary(...) = group_summary(<metric>, ...)
+# <metric>_group_summary(...) = group_summary(metric, ...)
 
-# Transformation functions returning scalars
+# Transformation functions returning scalars (the definitions are below)
 difference_from_summary(summary)
 ratio_from_summary(summary)
 group_min_from_summary(summary)
 group_max_from_summary(summary)
 
-# Metric-specific functions returning group summary and scalars
+derived_metric(metric, transformation, y_true, y_pred, *, sensitive_features, **other_kwargs) [TODO]
+# return a metric derived from the provided `metric`, where `metric` has the signature
+#   * metric(y_true, y_pred, **other_kwargs)
+# and `transformation` is a string 'difference', 'ratio', 'group_min' or 'group_max'.
+#
+# Alternatively, `metric` can be a metric group summary function, and `transformation`
+# can be a function one of the functions <transformation>_from_summary.
+
+make_derived_metric(metric, transformation) [TODO]
+# return a callable object <metric>_<transformation>:
+# <metric>_<transformation>(...) = derived_metric(metric, transformation, ...)
+
+# Predefined metrics are named according to the following patterns:
 <metric>_group_summary(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_difference(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_ratio(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_group_min(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_group_max(y_true, y_pred, *, sensitive_features, **other_kwargs)
 ```
 
-*Transformations and transformation codes*
+### Definitions of transformation functions
 
-|transformation function|output|metric-specific function|code|aif360|
+|transformation function|output|derived metric name|code|aif360|
 |-----------------------|------|------------------------|----|------|
 |`difference_from_summary`|max - min|`<metric>_difference`|D|unprivileged - privileged|
 |`ratio_from_summary`|min / max|`<metric>_ratio`|R| unprivileged / privileged|

```

Comment:
Might want to add note about behaviour when `max==0`

Reply:
Good point. Done.